app/Http/Controllers/Api/UserController.php

```diff
@@ -2,83 +2,70 @@
 
 namespace App\Http\Controllers;
 
+use App\Http\Controllers\Api\Util;
+use App\Models\User;
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
 
 class UserController extends Controller
 {
-    /**
-     * Display a listing of the resource.
-     *
-     * @return \Illuminate\Http\Response
-     */
-    public function index()
+    
+    public function importStudents(Request $request)
     {
-        //
-    }
+        $file = public_path($request->get('data_url'));
 
-    /**
-     * Show the form for creating a new resource.
-     *
-     * @return \Illuminate\Http\Response
-     */
-    public function create()
-    {
-        //
-    }
+        $validator = Validator::make($request->all(), [
+            'data_url' => 'required',
+        ]);
 
-    /**
-     * Store a newly created resource in storage.
-     *
-     * @param  \Illuminate\Http\Request  $request
-     * @return \Illuminate\Http\Response
-     */
-    public function store(Request $request)
-    {
-        //
-    }
+        if ($validator->fails()) {
+            return response()->json(['error' => "data_url field is required"], 400);
+        }
 
-    /**
-     * Display the specified resource.
-     *
-     * @param  int  $id
-     * @return \Illuminate\Http\Response
-     */
-    public function show($id)
-    {
-        //
-    }
+        $customerArr = Util::csvToArray($file);
 
-    /**
-     * Show the form for editing the specified resource.
-     *
-     * @param  int  $id
-     * @return \Illuminate\Http\Response
-     */
-    public function edit($id)
-    {
-        //
+        $data = [];
+        $exists = [];
+        $newStudents = [];
+        
+        for ($i = 0; $i < count($customerArr); $i++) {
+            
+            $checkDuplicateMatricNumber = User::where(['matric_number' => $customerArr[$i]['matric_number']])->first();
+            
+            if($checkDuplicateMatricNumber == null){
+                $data[] = [
+                    'id' => $this->uuid(),
+                    'matric_number' => $customerArr[$i]['matric_number'],
+                    'first_name' => $customerArr[$i]['first_name'],
+                    'other_names' => $customerArr[$i]['other_names'],
+                    'gender' => $customerArr[$i]['gender'],
+                    'password' => $customerArr[$i]['password'],
```

Comment:
@bynalab can we add default password say 123456? Also wondering if its possible to force them to change their password on their first login.

Reply:
Yea, about the force change of password, i can find a way to notify the front-end about the redirection to change-password page or so, like a boolean or something, to say 'hey, this person needs to change his/her password' 
Think that will do, do you?

Reply:
Yes that works. Could be something like `defaultPassword: true|false`, in the case of true, they should be redirected back to the change password page.